app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt

```diff
@@ -96,9 +96,11 @@ class SettingsActivity : AppCompatActivity(),
         val hasInternet = connectionLiveData.value
         val connectionState = serverLiveData.value
 
-        if (hasInternet == null || !hasInternet) {
-            setNoInternet()
-        } else if (hasInternet == true) {
+        if (hasInternet == null) {
+            return
+        }
+
```

Comment:
Is something like this possible and allowed by detekt/ktlint? If so, this can make the if statement less nested.
```
 if (false == hasInternet) {
     setNoInternet()
     return
 }
```

Reply:
Maybe, but I don't care how detekt/ktlint interprets this code in particular.

The code was written to
1) do nothing and return when `hasInternet` is `null`
2) do *something* when it is not null

So in order to make the program more _structured_, it doesn't (over)use return statements after some work has been done. I have intentionally chosen this approach, since this way, there is a clear distinction between a routine check and doing actual work.

Reply:
Good to see you have thought about it.